_develop/keypress-on-entry.md

```diff
@@ -0,0 +1,114 @@
+---
+layout: develop
+title: Respond To Specific Keypress
+
+summary: As a follow up to the writing widgets tutorial, this page will go through how to extend an Entry widget and respond to a specific keypress in it. At the end of the tutorial, you will have created an Entry that responds to pressing the Enter/Return button on the keyboard.
+
+order: 120
+---
+
+## Respond To Specific Keypress In Entry
+---
+
+In the traditional sense, GUI programs have used callbacks to customize actions for widgets. Fyne does not expose inserting closure callbacks inside widgets, but it does not need to. The Go language is plenty extensible to make this work.
+
+Simply use Type Embedding and extend the widget, along with adding a callback field for the specific action.
+
+First create a new type struct, we will call it `EnterEntry`. Then embed `widget.Entry` and insert a callback field called `OnEnter`.
+
+```go
+type EnterEntry struct {
```

Comment:
The other examples don’t export custom types, should we stick to that, or update the others to match?

Reply:
It is good practice not to, I will correct it here to match the others.